src/app/containers/PageHandlers/withVariant/index.test.jsx

```diff
@@ -2,104 +2,80 @@ import React from 'react';
 import { Router, Route } from 'react-router-dom';
 import { render } from '@testing-library/react';
 import { createMemoryHistory } from 'history';
-import WithVariant from '.';
 import { frontPagePath } from '#app/routes/regex';
+import * as Cookies from '#contexts/UserContext/cookies';
+import withVariant from '.';
+
+const Component = () => <h1>This is the BBC.</h1>;
+const ComponentWithVariantRedirect = withVariant(Component);
+
+const testWithVariant = ({ service, expectedPath }) => {
+  const history = createMemoryHistory({
+    initialEntries: [`/${service}`],
+  });
+
+  expect(history.location.pathname).toEqual(`/${service}`);
+
+  render(
+    <Router history={history}>
+      <Route path={frontPagePath}>
+        <ComponentWithVariantRedirect />
+      </Route>
+    </Router>,
+  );
+
+  expect(history.location.pathname).toEqual(expectedPath);
+};
 
 describe('WithVariant', () => {
-  const Component = () => <h1>This is the BBC.</h1>;
-  const ComponentWithVariantRedirect = WithVariant(Component);
-  const getMatchProps = (service, path = null) => ({
-    path: path || frontPagePath,
-    params: {
-      service,
-    },
+  beforeEach(() => {
+    jest.clearAllMocks();
+    Cookies.getPreferredVariant = jest.fn();
+    Cookies.setPreferredVariant = jest.fn();
   });
 
   describe('service with no default variant', () => {
     it('should not redirect', () => {
       const service = 'news';
-      const match = getMatchProps(service);
-      const history = createMemoryHistory({
-        initialEntries: [`/${service}`],
-      });
-
-      expect(history.location.pathname).toEqual(`/${service}`);
-
-      render(
-        <Router history={history}>
-          <Route path="/:service">
-            <ComponentWithVariantRedirect match={match} />
-          </Route>
-        </Router>,
-      );
-
-      expect(history.location.pathname).toEqual('/news');
+      const expectedPath = '/news';
+
+      testWithVariant({ service, expectedPath });
+
+      expect(Cookies.getPreferredVariant).not.toHaveBeenCalled();
+      expect(Cookies.setPreferredVariant).not.toHaveBeenCalled();
```

Comment:
What is the significance of checking these functions are called? Is it to ensure that cookies are set? If so you can do that easily because we are in a jsdom environment with jest. Just import the js-cookie library and check for the cookie